pkg/helm/informers.go

```diff
@@ -0,0 +1,179 @@
+package helm
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/mitchellh/hashstructure"
+	"github.com/pkg/errors"
+	apptypes "github.com/replicatedhq/kots/pkg/app/types"
+	"github.com/replicatedhq/kots/pkg/appstate"
+	appstatetypes "github.com/replicatedhq/kots/pkg/appstate/types"
+	identitydeploy "github.com/replicatedhq/kots/pkg/identity/deploy"
+	identitytypes "github.com/replicatedhq/kots/pkg/identity/types"
+	"github.com/replicatedhq/kots/pkg/kotsutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	registrytypes "github.com/replicatedhq/kots/pkg/registry/types"
+	"github.com/replicatedhq/kots/pkg/render"
+	"github.com/replicatedhq/kots/pkg/template"
+	"github.com/replicatedhq/kots/pkg/util"
+	"k8s.io/client-go/kubernetes"
+)
+
+var monitorMap map[string]*appstate.Monitor
+
+func initMonitor(clientset kubernetes.Interface, targetNamespace string) {
+	if monitorMap == nil {
+		monitorMap = make(map[string]*appstate.Monitor)
+	}
+
+	if monitorMap[targetNamespace] == nil {
+		monitorMap[targetNamespace] = appstate.NewMonitor(clientset, targetNamespace)
+	}
+
+	go runAppStateMonitor(monitorMap[targetNamespace])
+}
+
+func resumeHelmStatusInformers() {
+	for _, appName := range GetCachedHelmApps() {
```

Comment:
I would change this to take app name as parameter and remove the loop since this is called when a secret is added.  This this comment will be true I think

> // apply the informers for this app since they havent been applied yet